db/migrate/20190801143210_convert_dns_conflict_timeout_setting.rb

```diff
@@ -0,0 +1,46 @@
+class FakeSetting < ApplicationRecord
+  self.table_name = 'settings'
+
+  def default=(v)
+    write_attribute :default, v.to_yaml
+  end
+
+  def value=(v)
+    v = v.to_yaml unless v.nil?
+    write_attribute :value, v
+  end
+
+  def value
+    v = read_attribute(:value)
+    YAML.load(v) unless v.nil?
+  end
+end
+
+class ConvertDnsConflictTimeoutSetting < ActiveRecord::Migration[5.2]
+  def clean_cache
+    Rails.cache.delete(Setting.cache_key("dns_conflict_timeout"))
+    Rails.cache.delete(Setting.cache_key("dns_timeout"))
+  end
+
+  def up
+    Setting.without_auditing do
+      old_setting = FakeSetting.find_by_name("dns_conflict_timeout")
+      new_setting = FakeSetting.find_by_name("dns_timeout")
+      return if old_setting.nil? || new_setting.nil?
+      # only migrate the value if it was redefined (default was 3)
+      if old_setting.value != 3
+        new_setting.value = [old_setting.value]
```

Comment:
Do we want to set this to `[old_setting.value / 2, old_setting.value / 2]` so there's at least one retry?

Reply:
I was thinking the same, then I preferred simplicity. It worked so it will probably work too. Neverless, I believe that majority of users never touched this setting - it's good as a workaround enabler.

Reply:
Can do another rebase if you insist.

Reply:
Any chance to get it in? I have it as a sprint task, many thanks!

Reply:
I don't care that much (as I said, :+1:). This can also be mentioned in the upgrade warnings. When the user changed the previous setting, they should always look at this new setting after upgrading. If you can mention that in an upgrade warning, then I'm fine with merging. I think @ares wants to test this so I'll leave the honors to him.

Reply:
I don't understand what do I need to warn about - the upgrade is seamless. Foreman will respect what user set explicitly, in case it was a default value it will convert it to more appropriate array. There is nothing to check.

Reply:
My interpretation of Upgrade Warnings is that it's things the user should be aware of when upgrading. Changing behavior is a great example of that and this PR does change behavior. Most users won't care, but those who do care about details, typically appreciate it.

@tbrisker does my definition of Upgrade Warnings match yours?

Reply:
I agree we should document this, saying something like "if you changed dns_conflict_timeout" setting before, you may want to revisit and modify the new value to something more appropriate for your environment, explaining the array format"

Reply:
Also that it has a new name to better cover what it's actually used for. That way users know to look for the right setting.

Reply:
Sure, where do we put Upgrade Warnings @tbrisker ? Is this like normal Upgrade Notes we have in the docs or are we talking about something different?

Reply:
The above link (https://github.com/theforeman/theforeman.org/blob/gh-pages/_includes/manuals/nightly/1.2_release_notes.md#upgrade-warnings) is the one I was thinking about, so indeed the regular Upgrade Notes.

Reply:
Found it here: https://theforeman.org/manuals/1.22/index.html#Upgradewarnings

Created: https://community.theforeman.org/t/release-notes-and-headline-features-for-1-24/14877

Reply:
We have nightly for staging the 1.24 release. 1.23 has already been branched into its own files.